### ISSUE TITLE
Previews still very slow after plugin upgrade

### ISSUE THREAD

**Issue by kilinkis** [prismicio/prismic-gatsby#510]
hi @angeloashmore, I think might be a follow up from #500, but we are still experiencing very long times after upgrading, even more than before.

Answering your questions from that Issue:

- **How many documents do you have in your repository?**
    About 6000 (about 300 per locale)
- **Is the long load time on the preview resolver page (usually /preview), or is it on the actual page being previewed?**
    This happens on the URL of the previewed document
- **In your network tab in your browser's DevTools, do you see any duplicate or looping requests?**
    On the network tab, I see that there are requests for all the documents of the current locale, and once that is done, nothing else happens in the Network tab, and the Loading Preview screen remains. Only after like 2 more minutes, the modal window clears and I can see my page with the new content coming from the preview.

Our versions are:
"gatsby-source-prismic": "5.2.7",
"gatsby-plugin-prismic-previews": "5.2.8",
"gatsby": "4.15.1",

Do you have any hints? thanks!

**Comment by kilinkis** [comment]
@angeloashmore pinging again in case this got lost

**Comment by angeloashmore** [comment]
Hi @kilinkis, thanks for the ping. I don't have an immediate fix I can recommend right now, but I am planning to make changes to the preview system that hopefully eliminates the issue you're describing.

More details are available in this RFC: https://github.com/prismicio/prismic-gatsby/issues/514

**Comment by angeloashmore** [comment]
This issue is stale and may be fixed in a recent version. Please update to the latest version and try again. If the issue continues to be a problem, please open a new issue. Thank you!

See the documentation for guides on updating to the latest versions: [Gatsby + Prismic Migration Guides](https://github.com/prismicio/prismic-gatsby/tree/main/docs#migration-guides)
